pandas/core/frame.py

```diff
@@ -835,15 +835,129 @@ def __len__(self):
 
     def dot(self, other):
         """
-        Matrix multiplication with DataFrame or Series objects
+        Find the dot product of two matrices.
+
+        Using two DataFrame, matrix, or Series objects -
+        this method finds the standard inner product (dot product)
+        of two conformable matrices. Matrices A and B are said
+        to be conformable for multiplication A(B) whenever the
+        number of columns in A = the number of rows in B.
+        (i.e. -  A is m × p and B is p × n)
 
         Parameters
         ----------
-        other : DataFrame or Series
+        other : DataFrame, matrix, or Series
+            Different combinations shown in examples below.
 
         Returns
         -------
-        dot_product : DataFrame or Series
+        dot_product : DataFrame, matrix, or Series
+            Return type depends on input, see examples below.
```

Comment:
One suggestion for clarification: "Return type depends on input" may imply that the return type depends on input parameter `other` more than on the object it is applied on, but we actually see behavior like
  DataFrame.dot(matrix) --> DataFrame
from your examples.